src/runtime/virtcontainers/kata_agent.go

```diff
@@ -911,10 +915,41 @@ func (k *kataAgent) constrainGRPCSpec(grpcSpec *grpc.Spec, passSeccomp bool, str
 	// Disable SELinux inside of the virtual machine, the label will apply
 	// to the KVM process
 	if grpcSpec.Process.SelinuxLabel != "" {
-		k.Logger().Info("SELinux label from config will be applied to the hypervisor process, not the VM workload")
+		k.Logger().Info("SELinux label from spec config will be applied to the hypervisor process, not the VM workload")
 		grpcSpec.Process.SelinuxLabel = ""
 	}
 
+	// This is a hack. The SELinux MLS is determined by the high-level
+	// runtime randomly and it is already applied to the container rootfs
+	// by the runtime. Hence, the agent needs to apply the same MLS to the
+	// container process. However, the MLS is overridden for VM-based containers
+	// by the runtime and it is different from the original MLS applied to the
+	// container rootfs. Fortunately, we can get the original MLS from the mount
+	// label in the spec and use it.
+	if !disableGuestSeLinux {
+		k.Logger().Info("SELinux label will be applied to the contrainer process inside guest")
+
+		var label string
+		if guestSeLinuxLabel != "" {
+			label = guestSeLinuxLabel
+		} else {
+			label = defaultSeLinuxContainerLabel
+		}
+
+		mountContext, err := selinux.NewContext(grpcSpec.Linux.MountLabel)
```

Comment:
Are the grpcSpec.Linux.MountLabel and grpcSpec.Process.SelinuxLabel ordinarily the same?
Feels like the need to have an additional field in the OCI spec to accomodate both the Process labels for the hypervisor and container process, if we were to do this cleanly in the long run.

Reply:
Sorry I misunderstood. We just need to change the type from `container_kvm_t` to `container_t`, so the mount label is no longer used. I updated the description about this in #4812.

> 2. Apply a security policy to a container
In the case of general containers, not VM-based containers, high-level container runtimes such as containerd label the security policy for containers (`container_file_t`, etc) with Multi-Category Security (MCS) on the container image and low-level container runtimes such as runc apply the policy (`container_t`, etc) with the same MCS to the container process.
> If we can pass the security policy provided the high-level runtime to the guest, the kata-agent can apply the policy to the containers inside the guest easily. However, this needs an additional work because the SELinux type is overwritten as the new type (`container_kvm_t`) for VMM processes by the high-level runtime and the runtime passes the type to the kata-runtime. This behavior was added by [this commit](https://github.com/containerd/containerd/commit/d715d009061edf5ed0da5aa81fe7b6d2a6b3c10c) to add SELinux support for VMM processes by the https://github.com/kata-containers/runtime/pull/2503. Hence, we need to go back from the `container_kvm_t` to the `container_t` on the kata-runtime side.